src/google/protobuf/compiler/cpp/cpp_helpers.cc

```diff
@@ -176,6 +176,18 @@ string FieldName(const FieldDescriptor* field) {
   return result;
 }
 
+string EnumName(const EnumValueDescriptor* enum_value, bool should_mangle) {
+  string result = enum_value->name();
+  if (should_mangle) {
+    string lower = enum_value->name();
+    LowerString(&lower);
```

Comment:
Updated the match to be case sensitive and added a note in the header file about optional mangling.